### PR TITLE
AHOYAPPS-572: Implement most of audio switch unit tests

### PR DESCRIPTION
**Pull request by Alton09**
## Description

Implement most of the audio switch library unit tests with 79% total line coverage.

## Breakdown

- Abstract all bluetooth code into its own package
- Abstract AudioManager logic into the AudioDeviceManager class
- Abstract wired headset broadcast receiver into it's own class
- Temporarily add CI job to run library unit tests
- Add unit tests with 79% total line coverage
- Upgrade kotlin plugin
- Use Pixel 3 instead of Pixel 4 for FTL tests
- Fix JavaDoc issues after converting files to kotlin

## Validation

- Pass CI
- Manually validate the library integration with the app

## Additional Notes

N/A

## Submission Checklist
 - [x] The source has been evaluated for semantic versioning changes and are reflected in ```versionName``` under `app/build.gradle`
 - [x] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code

### CODE REVIEW

File: audioswitch/src/test/java/com/twilio/audioswitch/bluetooth/BluetoothControllerTest.kt [hunk 1]
```diff
@@ -4,57 +4,59 @@ import android.bluetooth.BluetoothAdapter
...
 import com.nhaarman.mockitokotlin2.verify
-import com.twilio.audioswitch.LogWrapper
+import com.twilio.audioswitch.AudioDeviceManager
+import com.twilio.audioswitch.AudioFocusRequestWrapper
+import com.twilio.audioswitch.android.BuildWrapper
+import com.twilio.audioswitch.android.LogWrapper
 import org.junit.Test
 
-class BluetoothControllerTest {
+class BluetoothCaontrollerTest {
```
[the remark this answers is not in the document]
Comment (Alton09) [reply audioswitch/src/test/java/com/twilio/audioswitch/bluetooth/BluetoothControllerTest.kt:15]: Thanks for the review!

File: audioswitch/src/main/java/com/twilio/audioswitch/AudioDeviceSelector.kt [hunk 2]
```diff
@@ -1,79 +1,107 @@
...
+    }
+
+    companion object {
+
+        /**
+         * Constructs a new AudioDeviceSelector instance.
+         *
+         * @param context the application context
+         */
+        fun newInstance(context: Context): AudioDeviceSelector {
```
Comment (aaalaniz) [inline audioswitch/src/main/java/com/twilio/audioswitch/AudioDeviceSelector.kt:89]: What is the reason for the static creation method versus a constructor?

Comment (Alton09) [reply audioswitch/src/main/java/com/twilio/audioswitch/AudioDeviceSelector.kt:89]: So I initially tried implementing two constructors, one internal and one public. It resulted in code that looked something like this:

```
    internal constructor(
            logger: LogWrapper,
            audioDeviceManager: AudioDeviceManager,
            wiredHeadsetReceiver: WiredHeadsetReceiver,
            bluetoothController: BluetoothController?) {
        this.logger = logger
        this.audioDeviceManager = audioDeviceManager
        this.wiredHeadsetReceiver = wiredHeadsetReceiver
        this.bluetoothController = bluetoothController
    }

    constructor(context: Context) {
        val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
        val logger = LogWrapper()
        val audioDeviceManager =
                AudioDeviceManager(context,
                        logger,
                        audioManager,
                        BuildWrapper(),
                        AudioFocusRequestWrapper())
        this.logger = logger
        this.audioDeviceManager = audioDeviceManager
        this.wiredHeadsetReceiver = WiredHeadsetReceiver(context, logger)
        this.bluetoothController = BluetoothController.newInstance(context, logger, audioDeviceManager)
    }

    private val logger: LogWrapper
    private val audioDeviceManager: AudioDeviceManager
    private val wiredHeadsetReceiver: WiredHeadsetReceiver
    private val bluetoothController: BluetoothController?
```

Unfortunately you have to duplicate the property setter calls and it becomes more verbose. When you use the static factory it allows you to invoke the class's primary constructor and you define the declaration and assignment of the properties all in one place, versus the method above which assigns them in two different constructors along with having to explicitly declare the properties outside of the constructors.

Comment (Alton09) [reply audioswitch/src/main/java/com/twilio/audioswitch/AudioDeviceSelector.kt:89]: And the reason we need two constructors is to allow our tests to inject their own collaborators to the class under test so we can assert/verify their behavior.

**Pull request merged**